sstables/sstable_set.cc

```diff
@@ -325,6 +325,13 @@ std::unique_ptr<sstable_set_impl> partitioned_sstable_set::clone() const {
     return std::make_unique<partitioned_sstable_set>(_schema, _unleveled_sstables, _leveled_sstables, _all, _all_runs, _use_level_metadata);
 }
 
+future<> partitioned_sstable_set::clear_gently() noexcept {
+    co_await utils::clear_gently(_unleveled_sstables);
+    co_await utils::clear_gently(_leveled_sstables);
+    co_await utils::clear_gently(_all);
+    co_await utils::clear_gently(_all_runs);
+}
```

Comment:
Why are these needed? How large are sstable sets?

Reply:
We didn't see any stalls in testing but I'm afraid that in real-life they could grow to 10's of thousands of sstables.
Keep in mind that with ICS with 1GB default fragment size, a 1TB table holds a 1000 sstables.
With tablets this should be more controlled for sure.

Reply:
Adding `cleanup()` and the `clear_gently` calls isn't instrumental to exception safety, so I can postpone that to a follow-up after proving why and when it's needed.